data/tools/wesnoth_addon_manager

```diff
@@ -14,132 +14,141 @@ from subprocess import Popen
 import wesnoth.wmlparser3 as wmlparser
 from wesnoth.campaignserver_client import CampaignClient
 
+# This is the validation code for the -u arguments. It checks if the input path is valid
+def valid_file_path(path):
+	if os.path.isdir(path) or os.path.isfile(path):
+		return path
+	else:
+		sys.stderr.write(f'No such file or directory: {path}')
```

Comment:
f-strings are [new in Python 3.6](https://docs.python.org/3/reference/lexical_analysis.html#f-strings). Do we support Python 3.5 or older?

Reply:
Yeah knowing the target Python env would be good to know, do tell me.

Thanks!

Reply:
So I don't know our actual minimum Python version requirement (perhaps @Elvish-Hunter knows?) but what I can tell you for sure is that `wesnoth_addon_manager` must be able to run on Python 3.5 at the very minimum since that's the version running on wesnoth.org, which uses WAM for the add-ons web directory in particular.

Reply:
Ok I've noted all the modifications needed, I'm on it lated today as soon as possible.

Thanks!